src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotUseReferenceEqualsWithValueTypesTests.cs

```diff
@@ -341,9 +341,10 @@ End Class
         [Fact]
         public async Task Comparer_ReferenceTypesAreOK()
         {
-            await VerifyCS.VerifyAnalyzerAsync(
-                ReferenceAssemblies.NetCore.NetCoreApp50,
-                @"
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
```

Comment:
Isn't this something that will be needed by majority of new tests that will be added? I am wondering if we should create a common helper such as `VerifyAnalyzerWithNetCoreApp50Async` or such. I will let @bartonjs chime in here.

Reply:
I think that a lot of the new analyzers are going to end up with the minimum platform reference assembly requirements.

Without fully understanding things, this either seems common (and should stay as a shared helper) or "nothing is common" and the helper should be fully deleted.

Reply:
We have three main options:

1. We could update the default reference assemblies to use `NetCoreApp50` instead of the current default.
1. If the analyzers are only expected to work on .NET 5, we can put the tests for those analyzers in a package that runs on `net5` and update `AdditionalMetadataReferences.Default` to use `NetCoreApp50` for this case. The code fix verifiers use this by default.
1. We could create a new type like `CSharpCodeFixVerifier<TAnalyzer, TCodeFix>` which is the new core verifier type for the new tests (similar to how we have separate a verifier for testing security rules).

Reply:
I like 2/ as it goes in the line of what is (will be) done for the support of netcoreapp3.1